packages/credentials/src/issuer.ts

```diff
@@ -134,3 +144,64 @@ export async function issue({
       )
   }
 }
+
+/**
+* Revokes a Kilt credential on the blockchain, making it invalid.
+* 
+* @param attester The DID URL of the attester revoking the credential
+* @param submitterAccount The account that will submit and pay for the transaction
+* @param signCallback Callback function to sign the DID authorization 
+* @param credential The Verifiable Credential to be revoked
+* @returns An object containing:
+* - success: Boolean indicating if revocation was successful
+* - error?: Array of error messages if revocation failed
+* - info: Object containing blockchain transaction details:
+*   - blockNumber?: Block number where transaction was included
+*   - blockHash?: Hash of the block
+*   - transactionHash?: Hash of the transaction
+*/
+export async function revoke(
+  attester: DidUrl,
+  submitterAccount: KeyringPair,
+  signCallback: any,
+  credential: VerifiableCredential,
+ ): Promise<{success: boolean, error?: Array<string>, info: {blockNumber?: string, blockHash?: string, transactionHash?: string}}> {
+  try {
+    const api = Kilt.ConfigService.get('api')
+    const rootHash = credential.id?.split(':').pop()!
+    const decodedroothash = base58Decode(rootHash);
+    
+    const revokeTx = api.tx.attestation.revoke(decodedroothash, null) as any
+    const [submitter] = (await Kilt.getSignersForKeypair({
+          keypair: submitterAccount,
+          type: 'Ed25519',
+    })) as Array<SignerInterface<'Ed25519', KiltAddress>>
+ 
+    const authorizedTx = await authorizeTx(
+      attester,
+      revokeTx,
+      signCallback,
+      submitter.id
+    )
+    
+    const response = await KiltChain.Blockchain.signAndSubmitTx(authorizedTx, submitterAccount) as Object
+    const responseObj = JSON.parse(JSON.stringify(response));
+ 
+    return {
+      success: true,
+      info: {
+        blockNumber: responseObj.blockNumber,
+        blockHash: responseObj.status.finalized,
+        transactionHash: responseObj.txHash
+      }
+    }
+    
+  } catch (error: unknown) {
+    const errorMessage = error instanceof Error ? error.message : 'Unknown error occurred';
+    return {
+      success: false,
+      error: [errorMessage],
+      info: {}
+    }
+ }
```

Comment:
In order to make sure we can support future credential- and proof types without breaking the function interfaces, this function should be set up to detect what kind of revocation status method we are dealing with, and if it's a of the `KiltRevocationStatusV1` type, call the right logic to revoke the credential.

Can you move the implementation to `packages/credentials/src/V1/KiltRevocationStatusV1.ts`, and in the function here only check which type of credentialStatus method we are dealing with, call the function implemented in KiltRevocationStatusV1.ts if it's the right one, and error if not? the `issue` function above should give you an idea.

Btw all the info you need for revocation is contained in the credentialStatus id, which is composed as follows:
`${chainId}/kilt:attestation/${base58Encode(rootHash)}`
Because we have the chain id here as well, which contains the genesis hash of the chain on which the credential was attested, you can even check if we are connected to the right network (i.e., peregrine vs spiritnet)! See the `check` function in KiltRevocationStatusV1.ts for an example.

Reply:
I am working on a second commit for this.

Reply:
Fixed at: [110a5cc](https://github.com/KILTprotocol/sdk-js/pull/918/commits/110a5cc482545b61bf5e514493616e2ffd3aaa69)